packages/docgen/test/get-type-annotation.js

```diff
@@ -196,16 +196,12 @@ describe( 'Type annotations', () => {
 	describe( 'missing types', () => {
 		const node = getMissingTypesNode();
 
-		it( 'should throw an error if there is no return type', () => {
-			expect( () => getTypeAnnotation( returnTag, node, 0 ) ).toThrow(
-				"Could not find return type for function 'fn'."
-			);
+		it( 'should return null if there is no return type', () => {
+			expect( getTypeAnnotation( returnTag, node, 0 ) ).toBeNull();
 		} );
 
-		it( 'should throw an error if there is no param type', () => {
-			expect( () => getTypeAnnotation( paramTag, node, 0 ) ).toThrow(
-				"Could not find type for parameter 'foo' in function 'fn'."
-			);
+		it( 'should return null if there is no param type', () => {
+			expect( getTypeAnnotation( paramTag, node, 0 ) ).toBeNull();
 		} );
```

Comment:
Can we add some tests for the case where it will throw? That's still a branch in the code.

Reply:
it seems like it might be worth removing the `try/catch` entirely because it shouldn't crash after fixing the null-dereference bug.

Reply:
Okay, in that case let's just remove that try/catch altogether as you suggested. If the code fails there then we'll hopefully get a nice bug report with a more descriptive error than the one we're currently spitting out anyway 🙂

Reply:
removed the try/catch and had to update other code. I did try a few types that I think should be totally invalid and they don't throw. I think the only thing that did throw was attempting to access the `typeAnnotation` assuming it was there, then trying to grab the name identifier on a function that doesn't exist _for the sake of error reporting_. without these I think we're safe apart from unlikely bugs that could always creep up